manifests/package.pp

```diff
@@ -12,9 +12,18 @@
     }
   } else {
     package { $forge_server::package:
-      ensure   => present,
+      ensure   => $::forge_server::pkg_ensure,
       provider => $::forge_server::provider
     }
   }
 
+  # If we are ensuring the latest gem package, uninstall old versions
+  if $::forge_server::pkg_ensure == 'latest' {
```

Comment:
Let's shelf the cleanup feature for now, then. After re-evaluating, I, personally, don't think that it would add a whole lot of value. The reason being that I can only think of one use case for this (which was the original reason why I decided to try to implement it in the first place).

I originally did this to handle the use case where the latest version is installed, but a previous version is 'ensured'; that scenario leads to both versions being installed but only the later version useful. 

Alternatively, we can handle this when executing the gem (I believe that would be in the template? Correct me if I'm wrong). According to [some gem documentation I found] (http://stackoverflow.com/questions/4373128/how-do-i-activate-a-different-version-of-a-particular-gem), you can run a specific version of the gem by doing the following:
```
<gem-binary> _<gem-version>_

e.g. puppet-forge-server _1.7.3_
```

Maybe we can add something to the template? It might be as easy as (pending testing, of course):
```
<%- if @pkg_ensure != "latest" && @pkg_ensure != "present" then -%>
prog="<%= @package %> _<%= @pkg_ensure %>_"
LOCKFILE=/var/lock/subsys/<%= @package %>
<%- else -%>
prog="<%= @package %>"
LOCKFILE=/var/lock/subsys/$prog
<%- end -%>
```

Thoughts?